### PR TITLE
feat: add explanation with category in the Nutri-Score and Eco-Score knowledge panels when not applicable

### PR DESCRIPTION
**Pull request by stephanegigandet**
When a product does not have a Nutri-Score or an Eco-Score because the product category is excluded from the Nutri-Score and Eco-Score, we now explain why in the corresponding knowledge panel, and we list the category.

image

Fixes https://github.com/openfoodfacts/openfoodfacts-server/issues/6092 and https://github.com/openfoodfacts/openfoodfacts-server/issues/6774

### CODE REVIEW

File: lib/ProductOpener/Attributes.pm [hunk 1]
```diff
@@ -656,14 +663,30 @@ sub compute_attribute_ecoscore($$$) {
 		}
 		$attribute_ref->{icon_url} = "$static_subdomain/images/attributes/ecoscore-$grade.svg";
 	}
+	# Eco-Score is not-applicable
+	elsif ((defined $product_ref->{ecoscore_grade}) and ($product_ref->{ecoscore_grade} eq "not-applicable")) {
+		$attribute_ref->{status} = "unknown";
+		$attribute_ref->{icon_url} = "$static_subdomain/images/attributes/ecoscore-unknown.svg";
```
Comment (alexgarel) [inline lib/ProductOpener/Attributes.pm:669]: you did not include the image, did you ?

File: lib/ProductOpener/Lang.pm [hunk 2]
```diff
@@ -210,14 +211,55 @@ If a translation is not available, the function returns English.
...
+
+e.g. "This is a string with {a_variable} and {another_variable}."
+
+Variables between { } are interpolated with the corresponding entry
+in the $variables_ref hash reference.
+
+If a translation is not available, the function returns English.
+
+=head3 Arguments
+
```
Comment (alexgarel) [inline lib/ProductOpener/Lang.pm:245]:
```suggestion

 =head4 target language $target_lc
 
```

**Review by alexgarel — CHANGES_REQUESTED** [review]
LGTM but you forgot an image. Also I was expecting a new nutriscore-not-applicable logo ;-)

**Comment by alexgarel** [comment]
Also, @stephanegigandet, could be good to add a test for eco-score not applicable…

**Comment by alexgarel** [comment]
proposal for logo:
nutriscore-notapplicable

**Comment by stephanegigandet** [comment]
@alexgarel for the logos, I made a separate PR:
https://github.com/openfoodfacts/openfoodfacts-server/pull/6848/files

Santé publique France has an official logo for "Nutri-Score unknown", so I think we should use it.

I think we can use the same logo for "Nutri-Score not-applicable" as well.

Same for Eco-Score unknown and not-applicable (yet), we can use the same logo.

**Comment by stephanegigandet** [comment]
> Also, @stephanegigandet, could be good to add a test for eco-score not applicable…

Good point, I added a test.

File: lib/ProductOpener/Attributes.pm (hunk 1 above)
Comment (alexgarel) [reply lib/ProductOpener/Attributes.pm:669]: will come with #6848

**Review by alexgarel — APPROVED** [review]
Good.

**Pull request merged**